cohere/responses/generation.py

```diff
@@ -114,3 +115,22 @@ def prompt(self) -> str:
         """Returns the prompt used as input"""
         return self[0].prompt  # should all be the same
 
+
+StreamingTokenLikelihood = NamedTuple("TokenLikelihood", [("index", int), ("token", str), ("likelihood", float)])
+
+
+class StreamingGenerations(CohereObject):
+
+    def __init__(self, response):
+        self.response = response
+
+    def __iter__(self) -> Generator[StreamingTokenLikelihood, None, None]:
```

Comment:
I wonder if we should expose an iterator per generation, i.e. grouped by `StreamingTokenLikelihood.Index`

Reply:
not sure how 🤔

Reply:
like a separate method that returns a list of iterators

Reply:
There isn't really a good way to do this without consuming tokens or essentially not-streaming.